addons/pos_loyalty/tests/test_frontend.py

```diff
@@ -2238,3 +2238,63 @@ def test_apply_reward_on_product_scan(self):
             "PosRewardProductScanGS1",
             login="pos_admin",
         )
+
+    def test_coupon_pricelist(self):
+        tax01 = self.env["account.tax"].create({
+            "name": "C01 Tax",
+            "amount": "0.00",
+        })
+
+        self.env["product.product"].create(
+            {
+                "name": "Test Product 1",
+                "is_storable": True,
+                "list_price": 25,
+                "available_in_pos": True,
+                "taxes_id": [(6, 0, [tax01.id])],
+            }
+        )
+
+        alt_pos_config = self.main_pos_config.copy()
+        pricelist_1 = self.env['product.pricelist'].create(
+            {
+                "name": "test pricelist 1",
+                "currency_id": self.env.ref("base.USD").id,
+            }
+        )
+        alt_pos_config.write({
+            'use_pricelist': True,
+            'available_pricelist_ids': [(4, pricelist_1.id)],
+            'pricelist_id': pricelist_1.id,
+        })
+
+        self.env["loyalty.program"].create(
+            {
+                "name": "Test Loyalty Program",
+                "program_type": "promotion",
+                "trigger": "with_code",
+                'pos_ok': True,
+                "pricelist_ids": [(4, pricelist_1.id)],
+                "rule_ids": [
+                    (0, 0, {"mode": "with_code", "code": "hellopromo", "minimum_amount": 10}),
+                ],
+                "reward_ids": [
+                    (0, 0, {
+                        "reward_type": "discount",
+                        "discount": 10,
+                        "discount_mode": "percent",
+                        "discount_applicability": "order",
+                        "required_points": 1,
+                    }),
+                ],
+                'pos_config_ids': [Command.link(alt_pos_config.id)],
+
+            }
+        )
+
+        alt_pos_config.with_user(self.pos_admin).open_ui()
+        self.start_tour(
+            "/pos/web?config_id=%d" % alt_pos_config.id,
+            "PosLoyaltyPromocodePricelist",
+            login="pos_admin",
```

Comment:
Always try to use `login="pos_user"` this way we can detect more access error bugs as the pos_user has the minimum access right possible for the PoS